profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/CMakeLists.txt

```diff
@@ -192,4 +192,6 @@ add_library(${PROFILER_SHARED_LIB_NAME} SHARED
 set_target_properties(${PROFILER_SHARED_LIB_NAME} PROPERTIES PREFIX "")
 
 # Define linker libraries
-target_link_libraries(${PROFILER_SHARED_LIB_NAME} ${PROFILER_STATIC_LIB_NAME})
+target_link_libraries(${PROFILER_SHARED_LIB_NAME}
+    -Wl,--whole-archive $<TARGET_FILE:${PROFILER_STATIC_LIB_NAME}> -Wl,--no-whole-archive
```

Comment:
I don't understand what this change does, can you explain it?

Reply:
Oh yeah that was a pain, because there is no `.def` file on linux (other means but not helpful)
Currently we build a static lib, and then build a shared library which links to the static one.
If you have functions that are called outside (ex.: PInvoke) and built in the static library. When creating the shared library, those functions won't be global (publicly visible).
This line says `include all object files into the shared library` and the functions (needed for PInvoke) will be visilble (same option on visual studio https://docs.microsoft.com/en-us/cpp/build/reference/wholearchive-include-all-library-object-files?view=msvc-170)
This has a significant hit in the linking time, but this is temporary. I will revisit the way we compile the profiler.